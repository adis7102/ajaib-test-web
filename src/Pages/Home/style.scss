@import '../../styles/fonts';
@import '../../styles//colors';

.home {
  display: grid;
  place-items: center;

  .subtitle {
    color: $color-N500;
    margin-top: 15px;
    margin-bottom: 25px;
  }

  .home-button { 
    padding: 18px;
    background-color: $color-alice-blue;
    border-radius: 5px;
    cursor: pointer;

    p {
      color: white;
      margin: 0;
    }

    &:hover {
      background-color: white;
      border: 1px solid black;

      p {
        color: black;
      }
    }
  }
}