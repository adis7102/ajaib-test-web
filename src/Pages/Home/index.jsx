import React from "react";
import { useNavigate } from "react-router-dom";

import Button from '../../Components/Button/index.jsx';

import "./style.scss";

const Home = () => {
  const navigate = useNavigate();

  const handleClick = () => {
    navigate("/datalist");
  };
  
  return (
    <div className="home">
      <h1>Hi, Welcome!</h1>
      <p className="subtitle">Click button below to continue!</p>
      <Button title="Click Here!" onClick={handleClick} padding={25} />
    </div>
  );
};

export default Home;
