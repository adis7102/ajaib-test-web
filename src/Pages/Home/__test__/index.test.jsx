import React from "react";
import Home from "../index.jsx";
import { mount } from "enzyme";
import { BrowserRouter as Router } from "react-router-dom";

const mockUseNav = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockUseNav,
}));

describe("Home Test", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <Router>
        <Home />
      </Router>
    );
  });

  it("Should render Home correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should click the button and move page to /detailist", () => {
    const button = wrapper.find(".button");

    button.simulate("click");

    expect(mockUseNav).toHaveBeenCalledWith('/datalist')
  });
});
