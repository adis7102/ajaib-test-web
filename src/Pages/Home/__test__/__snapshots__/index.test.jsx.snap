// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Test Should render Home correctly 1`] = `ReactWrapper {}`;
